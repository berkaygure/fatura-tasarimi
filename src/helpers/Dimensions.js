export const PER_PIXEL = 32;