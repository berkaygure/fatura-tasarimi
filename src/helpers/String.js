// @flow
// eslint-disable-next-line import/prefer-default-export
export const capitalize = (s: string): string => {
  if (typeof s !== 'string') return '';
  return s.charAt(0).toUpperCase() + s.slice(1);
};
