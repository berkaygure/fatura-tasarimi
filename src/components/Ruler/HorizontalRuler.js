import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { PER_PIXEL } from '../../helpers/Dimensions';
import './Ruler.css';

class HorizontalRuler extends Component {
  renderTicks() {
    const HRulerTicks = [];
    for (let i = 0; i < Math.ceil(this.props.width / PER_PIXEL); i += 1) {
      HRulerTicks.push(
        <div key={i} className="tick" style={{ width: PER_PIXEL, left: PER_PIXEL * i }}>
          <div className="rulerHNumber">{i}</div>
        </div>
      );
    }

    return HRulerTicks;
  }

  render() {
    return (
      <div className="ruler ruler-h">
        {this.renderTicks()}
        {this.props.followerW && this.props.followerX ? (
          <div
            className="ruler-follower-h"
            style={{
              left: this.props.followerX,
              width: this.props.followerW
            }}
          />
        ) : null}
      </div>
    );
  }
}

HorizontalRuler.propTypes = {
  width: PropTypes.number,
  followerW: PropTypes.number,
  followerX: PropTypes.number
};

export default HorizontalRuler;
