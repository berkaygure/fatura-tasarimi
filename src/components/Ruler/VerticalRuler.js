import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { PER_PIXEL } from '../../helpers/Dimensions';
import './Ruler.css';

class VerticalRuler extends Component {
  renderTicks() {
    const VRulerTicks = [];
    for (let i = 0; i < Math.ceil(this.props.height / PER_PIXEL); i++) {
      VRulerTicks.push(
        <div key={i} className="tick" style={{ height: PER_PIXEL, top: PER_PIXEL * i }}>
          <div className="rulerVNumber">{i}</div>
        </div>
      );
    }

    return VRulerTicks;
  }

  render() {
    return (
      <div className="ruler ruler-v">
        {this.renderTicks()}
        {this.props.followerY && this.props.followerH ? (
          <div
            className="ruler-follower-v"
            style={{
              top: this.props.followerY,
              height: this.props.followerH
            }}
          />
        ) : null}
      </div>
    );
  }
}

VerticalRuler.propTypes = {
  height: PropTypes.number,
  followerY: PropTypes.number,
  followerH: PropTypes.number
};

export default VerticalRuler;
