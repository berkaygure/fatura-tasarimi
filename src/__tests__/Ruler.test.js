import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import HorizontalRuler from '../components/Ruler/HorizontalRuler';
import { PER_PIXEL } from '../helpers/Dimensions';
import VerticalRuler from '../components/Ruler/VerticalRuler';

const PAPER_WIDTH = 672;
const PAPER_HEIGHT = 950;

describe('<HorizontalRuler />', () => {
  it('should be draw horizontal ruler', () => {
    const tree = renderer.create(<HorizontalRuler />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders tick `.tick`', () => {
    const wrapper = shallow(<HorizontalRuler width={PAPER_WIDTH} />);
    expect(wrapper.find('.tick')).toHaveLength(Math.ceil(PAPER_WIDTH / PER_PIXEL));
  });
});

describe('<VerticalRuler />', () => {
  it('should be draw vertical ruler', () => {
    const tree = renderer.create(<VerticalRuler />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders tick `.tick`', () => {
    const wrapper = shallow(<VerticalRuler height={PAPER_HEIGHT} />);
    expect(wrapper.find('.tick')).toHaveLength(Math.ceil(PAPER_HEIGHT / PER_PIXEL));
  });
});
