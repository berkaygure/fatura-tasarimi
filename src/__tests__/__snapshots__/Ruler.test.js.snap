// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HorizontalRuler /> should be draw horizontal ruler 1`] = `
<div
  className="ruler ruler-h"
/>
`;

exports[`<VerticalRuler /> should be draw vertical ruler 1`] = `
<div
  className="ruler ruler-v"
/>
`;
